modules/apps/transfer/keeper/relay.go

```diff
@@ -404,13 +460,45 @@ func createPacketDataBytesFromVersion(appVersion, sender, receiver, memo string,
 
 		token := tokens[0]
 		packetData := types.NewFungibleTokenPacketData(token.Denom.Path(), token.Amount, sender, receiver, memo)
-		packetDataBytes = packetData.GetBytes()
+
+		if err := packetData.ValidateBasic(); err != nil {
+			return nil, errorsmod.Wrapf(err, "failed to validate %s packet data", types.V1)
+		}
+
+		return packetData.GetBytes(), nil
 	case types.V2:
-		packetData := types.NewFungibleTokenPacketDataV2(tokens, sender, receiver, memo)
-		packetDataBytes = packetData.GetBytes()
+		// If forwarding is needed, move memo to forwarding packet data and set packet.Memo to empty string.
+		var forwardingPacketData types.ForwardingPacketData
+		if len(hops) > 0 {
+			forwardingPacketData = types.NewForwardingPacketData(memo, hops...)
+			memo = ""
+		}
+
+		packetData := types.NewFungibleTokenPacketDataV2(tokens, sender, receiver, memo, forwardingPacketData)
+
+		if err := packetData.ValidateBasic(); err != nil {
+			return nil, errorsmod.Wrapf(err, "failed to validate %s packet data", types.V2)
+		}
+
+		return packetData.GetBytes(), nil
 	default:
 		panic(fmt.Errorf("app version must be one of %s", types.SupportedVersions))
 	}
+}
+
+// burnCoin sends coins from the account to the transfer module account and then burn them.
+// We do this because bankKeeper.BurnCoins only works with a module account in SDK v0.50,
+// the next version of the SDK will allow burning coins from any account.
+// TODO: remove this function once we switch forwarding address to a module account (#6561)
+func (k Keeper) burnCoin(ctx sdk.Context, account sdk.AccAddress, coin sdk.Coin) error {
```

Comment:
Looks like this function can be removed and we can call `bankKeeper.BurnCoins` directly, right?

Reply:
https://github.com/cosmos/ibc-go/pull/6693/commits/cc46a63ea6e0644082b25761efbef0abfd759d91